### PR TITLE
python: update default propagation style headers [distributed tracing]

### PR DESCRIPTION
**Pull request by mabdinur**
## Motivation

Fixes failing parametric test: https://github.com/DataDog/system-tests/actions/runs/7787338188/job/21236589128

## Changes

With `ddtrace>=2.6.0` the default tracecontext header will be `datadog,tracecontext`.

## Workflow

1. ⚠️ Create your PR as draft ⚠️
2. Work on you PR until the CI passes (if something not related to your task is failing, you can ignore it)
3. Mark it as ready for review
    * Test logic is modified? -> Get a review from RFC owner. We're working on refining the `codeowners` file quickly.
    * Framework is modified, or non obvious usage of it -> get a review from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)

:rocket: Once your PR is reviewed, you can merge it!

🛟 [#apm-shared-testing](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X) 🛟

## Reviewer checklist

* [ ] [Relevant label](https://github.com/DataDog/system-tests/blob/main/docs/CI/labels.md) (`run-parametric-scenario`, `run-profiling-scenario`...) are presents
* [ ] No system-tests internal is modified. Otherwise, I have the approval from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)
* [ ] CI is green, or failing jobs are not related to this change (and you are 100% sure about this statement)
* [ ] A docker base image is modified?
    * [ ] the relevant `build-XXX-image` label is present
    * [ ] To R&P team: locally build and push the image to hub.docker.com 
* [ ] A scenario is added (or removed)?
    * [ ] Get a review from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)
    * [ ] Once merged, add (or remove) it in system-test-dasboard nightly

### CODE REVIEW

File: tests/parametric/test_headers_precedence.py [hunk 1]
```diff
@@ -486,7 +487,7 @@ def test_headers_precedence_propagationstyle_tracecontext(self, test_agent, test
     @missing_feature(context.library == "ruby", reason="library does not yet implement this default configuration")
     @irrelevant(context.library == "cpp", reason="library does not implement this default configuration")
     @irrelevant(context.library == "dotnet", reason="library does not implement this default configuration")
```
Comment (mabdinur) [inline tests/parametric/test_headers_precedence.py:489]:
```suggestions
    @irrelevant(context.library == "dotnet", reason="library does not implement this default configuration")
    @irrelevant(context.library == "golang", reason="library does not implement this default configuration")
```

File: tests/parametric/test_headers_precedence.py [hunk 2]
```diff
@@ -486,7 +487,7 @@ def test_headers_precedence_propagationstyle_tracecontext(self, test_agent, test
...
-    @irrelevant(context.library == "golang", reason="library does not implement this default configuration")
+    @irrelevant(context.library >= "python@2.6.0", reason="Default value was switched to datadog,tracecontext")
```
Comment (mabdinur) [inline tests/parametric/test_headers_precedence.py:490]:
```suggestion
    @irrelevant(context.library == "golang", reason="library does not implement this default configuration")
    @irrelevant(context.library >= "python@2.6.0", reason="Default value was switched to datadog,tracecontext")
```

**Review by robertomonteromiguel — CHANGES_REQUESTED** [review]
There are conflicts with the main branch

File: tests/parametric/test_headers_precedence.py [hunk 3]
```diff
@@ -218,6 +218,7 @@ def test_headers_precedence_propagationstyle_datadog(self, test_agent, test_libr
     @missing_feature(
         context.library == "ruby", reason="Issue: tracecontext,Datadog was never the default configuration"
     )
+    @irrelevant(context.library < "python@2.6.0", reason="Default value was switched to datadog,tracecontext")
```
Comment (mabdinur) [inline tests/parametric/test_headers_precedence.py:221]:
```suggestion
    @irrelevant(context.library <= "python@2.5.0", reason="Default value was switched to datadog,tracecontext")
```

**Review by cbeauchesne — APPROVED** [review]

**Review by robertomonteromiguel — APPROVED** [review]

File: tests/parametric/test_headers_precedence.py [hunk 4]
```diff
@@ -487,6 +488,7 @@ def test_headers_precedence_propagationstyle_tracecontext(self, test_agent, test
     @missing_feature(context.library == "ruby", reason="library does not yet implement this default configuration")
     @irrelevant(context.library == "cpp", reason="library does not implement this default configuration")
     @irrelevant(context.library == "dotnet", reason="library does not implement this default configuration")
+    @irrelevant(context.library >= "python@2.6.0", reason="Default value was switched to datadog,tracecontext")
```
Comment (mabdinur) [inline tests/parametric/test_headers_precedence.py:491]:
```suggestion
    @irrelevant(context.library < "python@2.6.0", reason="Default value was switched to datadog,tracecontext")
```

File: tests/parametric/test_headers_precedence.py (hunk 3 above)
Comment (mabdinur) [inline tests/parametric/test_headers_precedence.py:221]:
```suggestion
    @irrelevant(context.library >= "python@2.6.0", reason="Default value was switched to datadog,tracecontext")
```

**Pull request merged**